### PR TITLE
Fix Fabric API 0.86 incompatibility

### LINKED ISSUE #172: stopped working with fabric api 0.86.*
**Describe the bug**
dynamic light not available when you update to fabric 0.86.*

**To Reproduce**
1. install latest fabric api

**Expected behavior**

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Minecraft 1.20.1
 - Fabric Loader version0.14.21-1.20.1
 - Mods [e.g. fabric-api v0.86.1, lambdynamiclights v2.3.1

### PR DESCRIPTION
**Pull request by MUKSC**
Fix #172

### CODE REVIEW

File: src/main/java/dev/lambdaurora/lambdynlights/LambDynLightsCompat.java [hunk 1]
```diff
@@ -59,4 +59,14 @@ public static boolean isSodium05XInstalled() {
 			}
 		}).orElse(false);
 	}
+
+	public static boolean isFabricApi086OrLaterInstalled() {
+		return FabricLoader.getInstance().getModContainer("fabric-api").map(mod -> {
```
Comment (LambdAurora) [inline src/main/java/dev/lambdaurora/lambdynlights/LambDynLightsCompat.java]: This is not a good way to check if the fix needs to be applied, the mod should check for the Indigo module directly as some mod could bundle any Fabric API modules without requiring the whole of it.

File: src/main/java/dev/lambdaurora/lambdynlights/mixin/fabric/AoCalculatorMixin.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import net.minecraft.world.BlockRenderView;
+import org.spongepowered.asm.mixin.Dynamic;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Pseudo;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Pseudo
+@Mixin(targets = "net.fabricmc.fabric.impl.client.indigo.renderer.aocalc.AoCalculator", remap = false)
```
Comment (LambdAurora) [inline src/main/java/dev/lambdaurora/lambdynlights/mixin/fabric/AoCalculatorMixin.java:24]: I wonder if we can cheese the need for a mixin plugin with the power of optional injections. (This is a thing in mixin but it's not very used)

**Review by LambdAurora — CHANGES_REQUESTED** [review]

**Comment by MUKSC** [comment]
I tried without mixin plugin but it giving me error, so I added `require = 0` instead.
Is that ok?

**Comment by LambdAurora** [comment]
That should work, I'll do some experiments once I'm able to run the code.

**Review by LambdAurora — APPROVED** [review]
Tested with old and new versions, all working and no warning logs!

**Pull request merged**